libs/components/src/popover/vl-floating-ui.controller.ts

```diff
@@ -0,0 +1,195 @@
+        return [
+            // https://floating-ui.com/docs/offset
+            // offset() should generally be placed at the beginning of your middleware array.
+            offset(this.options.distance),
+            // https://floating-ui.com/docs/flip
+            flip(),
+            // https://floating-ui.com/docs/shift
+            shift(),
+            // https://floating-ui.com/docs/arrow
+            // arrow() should generally be placed toward the end of your middleware array, after shift().
+            arrow({ element: this.arrowElement, padding: 8 }),
+        ];
+    }
+
+    async updatePosition() {
+        const { x, y, strategy, placement, middlewareData } = await computePosition(
+            this.referenceElement,
+            this.floatingElement,
+            {
+                placement: this.options.placement,
+                middleware: this.middlewares,
+            }
+        );
+
+        Object.assign(this.floatingElement.style, {
+            position: strategy,
+            left: `${x}px`,
+            top: `${y}px`,
+        });
+
+        const staticSide = { top: 'bottom', right: 'left', bottom: 'top', left: 'right' }[placement.split('-')[0]];
+
+        if (middlewareData.arrow && this.arrowElement) {
+            const { x, y } = middlewareData.arrow;
+            Object.assign(this.arrowElement.style, {
+                left: x != null ? `${x}px` : '',
+                top: y != null ? `${y}px` : '',
+                [staticSide!]: `${-this.arrowElement.offsetWidth / 2}px`,
+            });
+        }
+    }
+
+    setOptions(options: FloatingControllerOptions) {
+        this.options = options;
+        return this;
+    }
+
+    handleResize(): void {
+        if (this.host.open) {
+            // ipv verbergen zouden we ook kunnen positie hercalculeren, maar dit impliceert performance decrease
+            // this.updatePosition(); // of met autoUpdate()
+            this.host.hide();
+        }
+    }
+
+    addEventListeners() {
+        this.referenceElement.addEventListener('keydown', this.handleKeyDown);
+
+        if (this.hasTrigger('click')) {
+            this.referenceElement.addEventListener('click', this.handleClick);
+            document.addEventListener('click', this.handleClickOutside, true);
+        }
+
+        if (this.hasTrigger('hover')) {
+            this.referenceElement.addEventListener('mouseover', this.handleMouseOver);
+            this.referenceElement.addEventListener('mouseout', this.handleMouseOut);
+        }
+
+        if (this.hasTrigger('focus')) {
+            this.referenceElement.addEventListener('focusin', this.handleFocusIn, true);
+            this.referenceElement.addEventListener('focusout', this.handleFocusOut, true);
+        }
+
+        window.addEventListener('resize', this.handleResize);
+    }
+
+    removeEventListeners() {
+        this.referenceElement?.removeEventListener('keydown', this.handleKeyDown);
+
+        if (this.hasTrigger('click')) {
```

Comment:
In principe kan je altijd een eventListener verwijderen ook al bestaat deze niet, je zou dit simpeler kunnen schrijven en de if statements weglaten.